lib/TH/generic/THVectorDefault.c

```diff
@@ -17,68 +17,100 @@ void THVector_(fill_DEFAULT)(real *x, const real c, const ptrdiff_t n) {
     x[i] = c;
 }
 
+void THVector_(cadd_DEFAULT)(real *z, const real *x, const real *y, const real c, const ptrdiff_t n)
+{
+  ptrdiff_t i = 0;
+
+  for(; i<n-4; i+=4)
```

Comment:
Can we get rid of this manual loop unrolling (everywhere)? I'd be shocked if on a modern compiler this actually helps. But not sure what compilers we're targeting here...